production/helm/loki/values.yaml

```
@@ -121,11 +121,14 @@ loki:
             period: 24h
     {{- end }}
 
-    {{- if or .Values.minio.enabled (eq .Values.loki.storage.type "s3") (eq .Values.loki.storage.type "gcs") }}
+    {{- if .Values.loki.rulerConfig}}
     ruler:
       storage:
+      {{- if or .Values.minio.enabled (eq .Values.loki.storage.type "s3") (eq .Values.loki.storage.type "gcs") }}
       {{- include "loki.rulerStorageConfig" . | nindent 4}}
-    {{- end -}}
+      {{- end }}
+    {{- toYaml .Values.loki.rulerConfig | nindent 2}}
+    {{- end }}
```

What would you think about something like this, so we'd always include that ruler storage section, and then by default we'd have the `rulerConfig` be an empty object? That would preserve parity with what we currently have. Might need to double check the indention is right in the suggestion.

```suggestion
    ruler:
      storage:
      {{- if or .Values.minio.enabled (eq .Values.loki.storage.type "s3") (eq .Values.loki.storage.type "gcs") }}
      {{- include "loki.rulerStorageConfig" . | nindent 4}}
      {{- end }}
    {{- with .Values.loki.rulerConfig}}
    {{- toYaml . | nindent 2}}
    {{- end }}
```